portafly/src/tests/components/shared/data-list/DataListTable.test.tsx

```diff
@@ -0,0 +1,304 @@
+import React from 'react'
+import { render } from 'tests/custom-render'
+import {
+  useDataListTable,
+  useDataListPagination,
+  useDataListFilters,
+  useDataListBulkActions
+} from 'components'
+import { DataListTable } from 'tests/components/shared/data-list/examples/DataListTable'
+import { fireEvent, within } from '@testing-library/react'
+import { sortable } from '@patternfly/react-table'
+
+jest.mock('components/shared/data-list/DataListContext')
+const useTableMock = useDataListTable as jest.Mock
+const usePaginationMock = useDataListPagination as jest.Mock
+const useFiltersMock = useDataListFilters as jest.Mock
+const useBulkActionsMock = useDataListBulkActions as jest.Mock
+
+const columns = [
+  {
+    categoryName: 'name',
+    title: 'Name',
+    transforms: [sortable]
+  },
+  {
+    categoryName: 'species',
+    title: 'Species / Race',
+    transforms: [sortable]
+  },
+  {
+    categoryName: 'nationality',
+    title: 'Nationality',
+    transforms: [sortable]
+  },
+  {
+    categoryName: 'state',
+    title: 'State',
+    transforms: [sortable]
+  }
+]
+
+const characters = [
+  {
+    id: 1,
+    name: 'Calanthe',
+    species: 'Human',
+    nationality: 'Cintra',
+    state: 'Dead'
+  },
+  {
+    id: 2,
+    name: 'Geralt',
+    species: 'Witcher',
+    nationality: '',
+    state: 'Alive'
+  },
+  {
+    id: 3,
+    name: 'Cirilla',
+    species: 'Human',
+    nationality: 'Cintra',
+    state: 'Alive'
+  },
+  {
+    id: 4,
+    name: 'Zoltan',
+    species: 'Dwarf',
+    nationality: 'Mahakam',
+    state: 'Alive'
+  }
+]
+
+const rows = characters.map((c) => ({
+  id: c.id,
+  cells: Object.keys(c).filter((k) => k !== 'id').map((k) => c[k]),
+  selected: false
+}))
+
+const isChecked = (c: HTMLElement) => (c as HTMLInputElement).checked
+
+const defaultTable = {
+  rows,
+  columns,
+  selectedRows: [],
+  sortBy: {},
+  setSortBy: jest.fn(),
+  selectOne: jest.fn(),
+  selectPage: jest.fn(),
+  selectAll: jest.fn()
+}
+const resetPagination = jest.fn()
+const defaultPagination = { startIdx: 0, endIdx: 5, resetPagination }
+const defaultFilters = { filters: {} }
+const defaultBulkActions = {}
+const resetMocks = () => {
+  useTableMock.mockReturnValue(defaultTable)
+  usePaginationMock.mockReturnValue(defaultPagination)
+  useFiltersMock.mockReturnValue(defaultFilters)
+  useBulkActionsMock.mockReturnValue(defaultBulkActions)
+}
+beforeAll(resetMocks)
+afterEach(resetMocks)
+
+const setup = () => {
+  const wrapper = render(<DataListTable />)
+  const getRow = (character) => wrapper.getByText(character.name).closest('tr') as HTMLElement
```

Comment:
I just discovered this is faulty (when filter is the same as name, it throws because it finds 2 elements).
```suggestion
  const table = within(wrapper.queryByRole('grid'}) as HTMLElement)
  const getRow = (character) => table.getByText(character.name).closest('tr') as HTMLElement
```